### PR TITLE
Add months possibility for repeating reminder feature (#3604)

### PR DESCRIPTION
**Pull request by vlo-rte**
- In release note :
  -  In chapter : Features
  -  Text : #3604 : Add months possibility for repeating reminder feature

### CODE REVIEW

File: services/cards-publication/src/main/modeling/swagger.yaml [hunk 1]
```diff
@@ -126,6 +126,11 @@ definitions:
         type: array
         items:
           type: integer
+      months:
+        description: Months of the year for the recurrence
+        type: array
+        items:
+          type: integer
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply services/cards-publication/src/main/modeling/swagger.yaml]: Done.

File: ui/main/src/app/modules/calendar/calendar.component.ts [hunk 2]
```diff
@@ -182,6 +183,19 @@ export class CalendarComponent implements OnInit, OnDestroy, AfterViewInit {
         return daysOfWeek;
     }
 
+    private getMonths(timeSpan: TimeSpan):Array<number>
+    {
+        let months = [];
+        if (timeSpan.recurrence) {
+            if (timeSpan.recurrence.months){
+                months = timeSpan.recurrence.months.map((d) => d % 12);
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/calendar/calendar.component.ts]: Deleted.

**Comment by freddidierRTE** [comment]
Add this test (which is passing) : 
```
    it('2020/08/29 Saturday 10:00 , locale = en , Recurrence : 16:30 / Wednesday(3), Friday(5) / August (7), October(9) => 2020/10/02 16:30 ', () => {
        moment.locale('en');
        const date = moment.tz('2020-08-29 10:00', 'Europe/Paris').valueOf();

        recurrence.hoursAndMinutes = {hours: 16, minutes: 30};
        recurrence.months = [7, 9];
        recurrence.daysOfWeek = [3,5];
        testCard.timeSpans = [new TimeSpan(0, null, recurrence)];

        const expectedResponseDate = moment.tz('2020-10-02 16:30', 'Europe/Paris').valueOf();
        const dateForRepeating = getNextTimeForRepeating(testCard, date);
        expect(new Date(dateForRepeating)).toEqual(new Date(expectedResponseDate));
    });

```

Ad the next one , which shall pass but is not passing : 

```
    it('2020/08/26 Wednesday 10:00 , locale = en , Recurrence : 16:30 / Tuesday(2) / August (7), october(9) => 2020/10/06 16:30 ', () => {
        moment.locale('en');
        const date = moment.tz('2020-08-26 10:00', 'Europe/Paris').valueOf();

        recurrence.hoursAndMinutes = {hours: 16, minutes: 30};
        recurrence.months = [7, 9];
        recurrence.daysOfWeek = [2];
        testCard.timeSpans = [new TimeSpan(0, null, recurrence)];

        const expectedResponseDate = moment.tz('2020-10-06 16:30', 'Europe/Paris').valueOf();
        const dateForRepeating = getNextTimeForRepeating(testCard, date);
        expect(new Date(dateForRepeating)).toEqual(new Date(expectedResponseDate));
    });
```

This is due to the fowllowing control because we end up having to add more than 7 days when changing month :+1: 

```
   nb_add++;
   if (nb_add > 7) { // in case we have an invalid recurrence daysOfWeek array
                  return -1;
     }

```

I suggest replacing this way of controlling the weekday by a more explicit method that `control` the validity of the daysOfWeek values in the array 
+ a control in the back (method validate in CardProcessingService)

**Comment by freddidierRTE** [comment]
image

Put the month list in a new line and put an intermediate line between the day of week list  and the month list

**Comment by freddidierRTE** [comment]
image

Separate month and day of week , each having its own box

File: ui/main/package.json [hunk 3]
```diff
@@ -27,9 +27,10 @@
     "@fortawesome/fontawesome-free": "6.2.0",
     "@fullcalendar/angular": "5.11.2",
     "@fullcalendar/bootstrap": "5.11.3",
-    "@fullcalendar/core": "5.11.3",
-    "@fullcalendar/daygrid": "5.11.3",
+    "@fullcalendar/core": "^5.11.3",
```
Comment (freddidierRTE) [inline ui/main/package.json:30]: Do not use ^ for version number as it does guarantee the version (we have to keep upgrades under control wuht precise version of dependencies)

File: ui/main/package.json [hunk 4]
```diff
@@ -27,9 +27,10 @@
...
+    "@fullcalendar/daygrid": "^5.11.3",
     "@fullcalendar/interaction": "5.11.3",
+    "@fullcalendar/rrule": "^5.11.3",
```
Comment (freddidierRTE) [inline ui/main/package.json:33]: Do not use ^ for version number as it does guarantee the version (we have to keep upgrades under control wuht precise version of dependencies)

File: ui/main/package.json [hunk 5]
```diff
@@ -56,6 +57,7 @@
     "ng-event-source": "1.0.14",
     "ngx-chips": "3.0.0",
     "ol": "6.5.0",
+    "rrule": "^2.7.1",
```
Comment (freddidierRTE) [inline ui/main/package.json:60]: Do not use ^ for version number as it does guarantee the version (we have to keep upgrades under control wuht precise version of dependencies)

File: ui/main/package.json (hunk 3 above)
Comment (vlo-rte) [reply ui/main/package.json:30]: Done.

File: ui/main/package.json (hunk 4 above)
Comment (vlo-rte) [reply ui/main/package.json:33]: Done.

File: ui/main/package.json (hunk 5 above)
Comment (vlo-rte) [reply ui/main/package.json:60]: Done.

**Comment by vlo-rte** [comment]
> image
> 
> Put the month list in a new line and put an intermediate line between the day of week list and the month list

Done.

**Comment by vlo-rte** [comment]
> image
> 
> Separate month and day of week , each having its own border box

Done.

**Comment by vlo-rte** [comment]
> if (nb_add > 7) { // in case we have an invalid recurrence daysOfWeek array

Done.

**Pull request merged**